test.sh

```diff
@@ -2,34 +2,54 @@
 
 import glob
 import os
-import subprocess
 import sys
+from subprocess import Popen, PIPE
+
+try:
+    from termcolor import cprint
+except ModuleNotFoundError:
+    print("Install module termcolor for better experience!")
+    print("python3 -m pip install --user termcolor")
+    def cprint(a, b, c):
+        print(a)
+
+def test_file(exe, input, output, error):
+    with open(output, 'rb') as o_:
+        out_expected = o_.read()
+
+    with open(error, 'rb') as e_:
+        err_expected = e_.read()
+
+    with open(input) as f_input:
+        p = Popen([exe], stdout=PIPE, stderr=PIPE, stdin=f_input)
+        out, err = p.communicate()
+
+    if p.returncode != 0 or out != out_expected or err != err_expected:
+        cprint("\u274C TEST {} FAILED".format(input), 'white', 'on_red')
+        return
+    
+    cprint("\u2713 TEST {} PASSED".format(input),'white','on_green')
+
+
+    with open(input) as f_input:
```

Comment:
Valgrind nie wykrywa memleaków, potrzeba jakiś flag np --leak-check=full --show-leak-kinds=all --errors-for-leak-kinds=all. Można jeszcze użyć  --leak-resolution lub --error-exitcode.